pilot/pkg/xds/ads.go

```diff
@@ -280,6 +281,20 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream discovery.AggregatedD
 func (s *DiscoveryServer) shouldRespond(con *Connection, request *discovery.DiscoveryRequest) bool {
 	stype := v3.GetShortType(request.TypeUrl)
 
+	// get health checks out of the way. we do not want to respond
+	// to health info
+	if request.TypeUrl == health.HealthInfoTypeURL {
+		event := HealthEvent{}
+		if request.ErrorDetail == nil {
+			event.Healthy = true
+		} else {
+			event.Healthy = false
+			event.Message = request.ErrorDetail.Message
+		}
+		s.InternalGen.UpdateWorkloadEntryHealth(con.proxy, event)
```

Comment:
Strongly suggest moving it out to `processRequest`. And make it running in a new goroutine to unblock the normal xds process

Reply:
Why do we want to move it out of `processRequest`? Where else would we move it? This seems like an appropriate place.

@howardjohn thoughts?

Reply:
I mean move it directly to `processRequest`. This should not be part of `shouldRespond`.